### PR TITLE
doc/examples: golang - list_nodes.go

### PR DESCRIPTION
**Pull request by dougsland**
Just an example how to use golang and bluechi.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

### CODE REVIEW

**Comment by dougsland** [comment]
I was just looking for test bluechi + golang and this should be enough until we have a dbus generation to a golang module as we have in python. 
/cc @engelmi

File: doc/bluechi-examples/golang/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+list_nodes
```
Comment (engelmi) [inline doc/bluechi-examples/golang/.gitignore]: The `.gitignore` here is not needed. Could you please remove it?

File: doc/bluechi-examples/golang/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# golang examples
```
Comment (engelmi) [inline doc/bluechi-examples/golang/README.md]: The content of this file should be appended to https://github.com/containers/bluechi/blob/main/doc/docs/api/examples.md as a new section
```
## GoLang

...
```

File: doc/bluechi-examples/golang/list_nodes.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
```
Comment (engelmi) [inline doc/bluechi-examples/golang/list_nodes.go]: The `doc/bluechi-examples` is intended for the bindings (only python currently) - maybe the directory name is misleading. Could you please move it to `doc/api-examples/go/list_nodes.go`?

File: doc/bluechi-examples/golang/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+``` bash
+# dnf install go-toolset
+```
+
+**Adding dbus library**:
+
+``` bash
+# go install github.com/godbus/dbus
+```
+
```
Comment (engelmi) [inline doc/bluechi-examples/golang/README.md]: Could you add another section here for showing the content of `list_nodes.go`? Then you can integrate the source file like this
```go
--8<-- "api-examples/go/list_nodes.go"
```

Maybe we need to add a new package to the readthedocs, but we can do that after this PR is merged.

**Review by engelmi — CHANGES_REQUESTED** [review]
I really like adding more languages to the list of examples on how to use the D-Bus API :) I have a few change requests, though.

**Comment by dougsland** [comment]
closing, rebase made a mess.

**Comment by dougsland** [comment]
the new one: https://github.com/containers/bluechi/pull/528

**Pull request closed without merging**